config-model/src/main/java/com/yahoo/vespa/model/container/search/ContainerSearch.java

```diff
@@ -178,9 +182,11 @@ public void getConfig(QrSearchersConfig.Builder builder) {
     		if (sys instanceof IndexedSearchCluster) {
     			scB.rowbits(sys.getRowBits());
                 for (Dispatch tld: ((IndexedSearchCluster)sys).getTLDs()) {
-                	scB.dispatcher(new QrSearchersConfig.Searchcluster.Dispatcher.Builder().
-                			host(tld.getHostname()).
-                			port(tld.getDispatchPort()));
+                    if (hostName.equals(tld.getHostname())) {
+                        scB.dispatcher(new QrSearchersConfig.Searchcluster.Dispatcher.Builder().
+                                host(tld.getHostname()).
```

Comment:
Now the hostname could be replaced with "localhost". Not sure if that is desirable or not.